assayist/processor/base.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from abc import ABC, abstractmethod
+import json
+import neomodel
+import os
+
+from assayist.common.models import content, source
+from assayist.processor.configuration import config
+from assayist.processor.logging import log
+
+
+class Analyzer(ABC):
+    """Base Abstract class that analyzers will inherit from."""
+
+    METADATA_DIR = '/metadata/'
+    MESSAGE_FILE = METADATA_DIR + 'message.json'
+    BUILD_FILE = METADATA_DIR + 'buildinfo.json'
+    TASK_FILE = METADATA_DIR + 'taskinfo.json'
+    RPM_FILE = METADATA_DIR + 'rpms.json'
+    ARCHIVE_FILE = METADATA_DIR + 'archives.json'
+    IMAGE_RPM_FILE = METADATA_DIR + 'image-rpms.json'
+    BUILDROOT_FILE = METADATA_DIR + 'buildroot-components.json'
+
+    def main(self):
+        """Call this to run the analyzer."""
+        neomodel.config.DATABASE_URL = config.DATABASE_URL
+        neomodel.config.AUTO_INSTALL_LABELS = True
```

Comment:
I think this requires more discussion. I was thinking we'd likely want to handle this manually or separately (i.e. running `neomodel_install_labels`). One of the issues is that if we change constraints in the middle of a run of another analyzer, it could cause that other analyzer run to fail. I think adding the constraints and indexes should be done with caution.

Reply:
I was mostly cargo-cult copying what you had somewhere else. If the AUTO_INSTALL_LABELS line is a problem I'm happy to delete it.